src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -379,7 +382,14 @@ void RealmCoordinator::do_get_realm(RealmConfig&& config, std::shared_ptr<Realm>
     // rerun_on_open was set
     if (subscription_function) {
         const auto current_subscription = realm->get_latest_subscription_set();
-        if ((current_subscription.version() == 0) || (rerun_on_open && db_opened_first_time)) {
+        const auto subscription_version = current_subscription.version();
+        const auto open_for_the_first_time =
+            first_time_open.has_value() ? first_time_open.value() | db_created : db_created;
```

Comment:
`.value()` requires a higher iOS deployment target than we currently have (and is just generally a bad idea anyway). Using `std::optional<bool>` isn't really doing anything useful here though, as `none` and `some(false)` have the exact same result and so it can just be a `bool`.

Reply:
Oh, thanks for the reminder. I keep forgetting about this. Let me revert to a simple bool.

Reply:
I think we usually dereference the optional values when needed.